react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import ReactDOM from "react-dom";
+
+class NewsletterForm extends React.Component{
+  constructor(props){
+    super(props);
+    this.state = { value: '' }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e){
+    this.setState({ value: e.target.value })
+  }
+
+  handleSubmit(e){
+    console.log(this.state)
+    e.preventDefault();
+  }
+
+  render(){
+    return(
+      <form onSubmit={this.handleSubmit}>
+        <label htmlFor="email">Email</label>
+        <input type="text" onChange={this.handleChange}/>
```

Comment:
one thing missing here, you need to add the `value` attribute to the input and and assign it the value you have stored in state. This makes sure that React maintains control of the input, and that state is determining what is being displayed.